### DISCUSSION TITLE
Window Borderless Resizable

### DISCUSSION THREAD

**Discussion by tilkinsc** [dotnet/Silk.NET#1873] [Ideas]
Has anyone got a working implementation of this?

I am trying to roll my own title bar using the client area, however, Silk.NET's borderless windows can't be resized with the (I guess lack of) borders. I'm also interested in shadows, so that the window can pop.

https://stackoverflow.com/questions/39731497/create-window-without-titlebar-with-resizable-border-and-without-bogus-6px-whit
https://stackoverflow.com/questions/32750525/x11-frameless-resizeable-windows
Not sure if this is possible on Mac

**Comment by Perksey** [comment]
I don't think I've seen anyone do this in the past, but previously when I've done something like this with WinForms it's quite easy to modify the Width/Height of the window based on mouse deltas. It likely won't be hard to implement. As for shadows, this probably will be trickier.

**Reply by tilkinsc** [reply]
True. I was able to implement everything I needed aside the resizing. I am absolutely not certain how glfw and sdl2 will handle the resizing. I'm not sure if the WinAPI has the ability to directionally resize windows. I took calculator as an example and when you drag a maximized window, it actually resizes the left border of the window until original size. This way you can never have you mouse send a mouse up on the close/min/max buttons. Pretty clever. The immediate solution is to both scale and position the window accordingly.
